danger/src/ChangelogChecker.ts

```diff
@@ -0,0 +1,257 @@
+function getTags(
+  entry: string
+): { packageName: string | typeof DEFAULT_CHANGELOG_ENTRY_KEY; type: ChangelogEntryType } | null {
+  const result = {
+    type: DEFAULT_ENTRY_TYPE,
+    packageName: DEFAULT_CHANGELOG_ENTRY_KEY,
+  };
+
+  const tags = entry.match(/\[[^\]]*\]/g);
+  if (!tags) {
+    return result;
+  }
+  if (tags.length > 2) {
+    return null;
+  }
+
+  for (const tag of tags) {
+    switch (true) {
+      case /\[[\s-_]*(bug)?[\s-_]*fix[\s-_]*\]/i.test(tag):
+        result.type = ChangelogEntryType.BUG_FIXES;
+        break;
+      case /\[[\s-_]*(new)?[\s-_]*feature(s)?[\s-_]*\]/i.test(tag):
+        result.type = ChangelogEntryType.NEW_FEATURES;
+        break;
+      case /\[[\s-_]*breaking[\s-_]*(change)?[\s-_]*\]/i.test(tag):
+        result.type = ChangelogEntryType.BREAKING_CHANGES;
+        break;
+      default:
+        result['packageName'] = tag.replace(/\[|\]/g, '').trim();
+    }
+  }
+
+  return result;
+}
+
+/**
+ * Get suggested changelog entries from PR.
+ *
+ * If PR doesn't contais `# Changelog` section then this method returns:
+ * {
+ *   [DEFAULT_CHANGELOG_ENTRY_KEY]: <title of this pr without tags>
+ * }
+ * Otherwise it tries to parse PR's body.
+ */
+function getChangelogSuggestionFromPR(): ChangelogEntries {
+  const changelogEntries = {
+    [DEFAULT_CHANGELOG_ENTRY_KEY]: {
+      type: DEFAULT_ENTRY_TYPE,
+      message: pr.title.replace(/\[.*\]/, '').trim(),
+    },
+  };
+
+  const changelogTag = pr.body.match(/#\schangelog(([^#]*?)\s?)*/i)?.[0]?.replace(/^-/, '');
+  if (changelogTag) {
+    changelogTag
+      .split('\n')
+      .slice(1)
+      .map(line => line.trim())
+      .filter(line => line.length > 0)
+      .forEach(line => {
+        const tags = getTags(line);
+        console.log({ tags });
+        if (!tags) {
+          warn(`Cound't parse tags from PR's body. Line: ${line}.`);
+          return;
+        }
+
+        if (tags.packageName) {
+          changelogEntries[tags.packageName] = {
+            type: tags.type,
+            message: line.replace(/\[.*\]/, '').trim(),
+          };
+        }
+      });
+  }
+
+  return changelogEntries;
+}
+
+/**
+ * Check if the changelog was modified.
+ * If `CHANGELOG.md` doesn't exist in provided package, it retunrs false.
+ */
+function wasChangelogModified(packageName: string, modifiedFiles: string[]): boolean {
+  const changelogPath = getPackageChangelogPath(packageName);
+
+  return (
+    modifiedFiles.includes(changelogPath) ||
+    !fs.existsSync(path.join(getExpoRepositoryRootDir(), changelogPath))
+  );
+}
+
+/**
+ * Add additional information, like PRs author and PRs link, to the entry.
+ */
+function addPRInfoToChangelogEntry(entry: string): string {
+  // We need to escape link in that way to display them in their raw form.
+  return `\\- ${entry} (\\[#<span/>${pr.number}](https:<span/>//github.com/expo/expo/pull/<span/>${pr.number}) by \\[@<span/>${pr.user.login}](https:<span/>//github.com/${pr.user.login}))`;
```

Comment:
Soo, we're obtaining for example sth like this:
```js
`\\- Added something to whatever module you liked. (\\[#<span/>45322](https:<span/>//github.com/expo/expo/pull/<span/>45322) by \\[@<span/>bbarthec](https:<span/>//github.com/bbarthec))`;
```
I don't really get from where these `<span/>`s are coming from 🤔 Mind explaining?

Reply:
I needed that to escape GitHub markdown - it will shorten links. 
For example, the PR link will be displayed as #<number> but I want to display the whole link. 

However, I don't need this anymore ;)